src/lib/transaction_snark/test/zkapps_examples/tokens/tokens.ml

```diff
@@ -0,0 +1,415 @@
+             ( fst @@ Async.Thread_safe.block_on_async_exn
+             @@ Zkapps_tokens.child_forest pk token_id subtree )
+        |> Zkapp_command.Call_forest.cons_tree Account_updates.mint
+        |> Zkapp_command.Call_forest.cons_tree Account_updates.initialize
+        |> Zkapp_command.Call_forest.cons
+             (Account_updates.deploy ~balance_change:(fee_to_create_signed 2))
+        |> test_zkapp_command ~fee_payer_pk:pk ~signers ~initialize_ledger
+             ~finalize_ledger
+      in
+      ignore account
+
+    let%test_unit "Initialize, mint, transfer recursive succeeds, ignores \
+                   non-token" =
+      let subtree =
+        []
+        |> Zkapp_command.Call_forest.cons
+             { Account_update.authorization = Control.Signature Signature.dummy
+             ; body =
+                 Zkapps_examples.mk_update_body (fst mint_to_keys)
+                   ~use_full_commitment:true
+                   ~balance_change:(int_to_amount (-5)) ~token_id:owned_token_id
+                   ~may_use_token:Parents_own_token
+             }
+             ~calls:
+               ( []
+               (* Delegate call, should be checked. *)
+               |> Zkapp_command.Call_forest.cons
+                    { Account_update.authorization =
+                        Control.Signature Signature.dummy
+                    ; body =
+                        Zkapps_examples.mk_update_body (fst mint_to_keys)
+                          ~use_full_commitment:true
+                          ~balance_change:(int_to_amount 2)
+                          ~token_id:owned_token_id
+                          ~may_use_token:Inherit_from_parent
+                    }
+                    ~calls:
+                      ( []
+                      (* Delegate call, should be checked. *)
+                      |> Zkapp_command.Call_forest.cons
+                           { Account_update.authorization =
+                               Control.Signature Signature.dummy
+                           ; body =
+                               Zkapps_examples.mk_update_body (fst mint_to_keys)
+                                 ~use_full_commitment:true
+                                 ~balance_change:(int_to_amount 2)
+                                 ~token_id:owned_token_id
+                                 ~may_use_token:Inherit_from_parent
+                           }
+                      (* Parents_own_token, should be skipped. *)
+                      |> Zkapp_command.Call_forest.cons
+                           { Account_update.authorization =
+                               Control.Signature Signature.dummy
+                           ; body =
+                               Zkapps_examples.mk_update_body (fst mint_to_keys)
+                                 ~use_full_commitment:true
+                                 ~balance_change:(int_to_amount (-15))
+                                 ~token_id:Token_id.default
```

Comment:
can use just `~token_id`, since that's in scope with `Token_id.default`; same pattern below